aioamqp_consumer/producer.py

```diff
@@ -126,8 +206,20 @@ def __init__(
             await self._disconnect()
             raise
 
+    async def exchange_delete(self, exchange_name, **kwargs):
+        try:
+            assert not self._closed, 'Cannot delete while closed'
+
+            await self._connect()
+
+            await self._exchange_delete(exchange_name, **kwargs)
+        except:  # noqa
+            await self._disconnect()
+            raise
+
     async def _disconnect(self):
         self._known_queues = set()
+        self._known_exchanges = set()
```

Comment:
maybe here we need to reset `_binded_queues `?

Reply:
Yes, you'r right.

Reply:
May be we have to delete queue from  self._known_queues on queue_delete too?

Reply:
makes sense